feat(editor): add named fixes for code actions

camchenry opened the pull request:
- related to https://github.com/oxc-project/oxc/issues/10038

This allows a `Fix` to have a message associated with it as well, similar to `RuleFix`. This is used by the editor to show code actions for a given lint error. For a start, I've added a message to the `no-debugger` rule and modified a test that shows this gets an improved code action message in VS Code.

camchenry commented:
> [!WARNING]
> <b>This pull request is not mergeable via GitHub because a downstack PR is open. Once all requirements are satisfied, merge this PR as a stack <a href="https://app.graphite.dev/github/pr/oxc-project/oxc/10203?utm_source=stack-comment-downstack-mergeability-warning" >on Graphite</a>.</b>
> <a href="https://graphite.dev/docs/merge-pull-requests">Learn more</a>

* **#10203** <a href="https://app.graphite.dev/github/pr/oxc-project/oxc/10203?utm_source=stack-comment-icon" target="_blank"><img src="https://static.graphite.dev/graphite-32x32-black.png" alt="Graphite" width="10px" height="10px"/></a> 👈 <a href="https://app.graphite.dev/github/pr/oxc-project/oxc/10203?utm_source=stack-comment-view-in-graphite" target="_blank">(View in Graphite)</a>
* **#10168** <a href="https://app.graphite.dev/github/pr/oxc-project/oxc/10168?utm_source=stack-comment-icon" target="_blank"><img src="https://static.graphite.dev/graphite-32x32-black.png" alt="Graphite" width="10px" height="10px"/></a>
* `main`
<hr />

## How to use the Graphite Merge Queue

Add either label to this PR to merge it via the merge queue:
- <kbd>0-merge</kbd> - adds this PR to the back of the merge queue
- <kbd>hotfix</kbd> - for urgent hot fixes, skip the queue and merge this PR next

You must have a Graphite account in order to use the merge queue. Sign up using [this link](https://app.graphite.dev/invite/github/oxc-project?ref=merge-queue-instructions-in-stack-comment&prId=10203).

<h2></h2>
<sub>

An organization admin has enabled the [Graphite Merge Queue](https://graphite.dev/docs/graphite-merge-queue) in this repository.

</sub>

<sub>

Please do not merge from GitHub as this will restart CI on PRs being processed by the merge queue.

</sub>

<h2></h2>

This stack of pull requests is managed by <a href="https://graphite.dev?utm-source=stack-comment"><b>Graphite</b></a>. Learn more about <a href="https://stacking.dev/?utm_source=stack-comment">stacking</a>.
<!-- Current dependencies on/for this PR: -->

Sysix commented on crates/oxc_linter/src/snapshots/eslint_no_debugger.snap:
```diff
@@ -6,4 +6,4 @@ source: crates/oxc_linter/src/tester.rs
    ·          ────────
    ╰────
-  help: Replace `debugger` with `{}`.
+  help: Replace the debugger statement with an empty block
```
For me, a block is surrounded by `{}`. Would prefer just the `REMOVE_DEBUGGER` message. 

Can add later a test for https://github.com/oxc-project/oxc/issues/9561 to find different messages

camchenry commented:
### Merge activity

* **Apr 3, 10:34 PM EDT**: The merge label '0-merge' was detected. This PR will be added to the [Graphite merge queue](https://app.graphite.dev/merges?org=oxc-project&repo=oxc) once it meets the requirements.

camchenry committed:
feat(editor): add named fixes for code actions (#10203)

- related to https://github.com/oxc-project/oxc/issues/10038

This allows a `Fix` to have a message associated with it as well, similar to `RuleFix`. This is used by the editor to show code actions for a given lint error. For a start, I've added a message to the `no-debugger` rule and modified a test that shows this gets an improved code action message in VS Code.
